diff -Naur a/src/ssl/rec_wri.cpp b/src/ssl/rec_wri.cpp
--- a/src/ssl/rec_wri.cpp	2018-02-05 15:41:49.018313394 +0100
+++ b/src/ssl/rec_wri.cpp	2018-02-05 15:53:14.660325737 +0100
@@ -16,7 +16,7 @@
 /**
 * Record_Writer Constructor
 */
-Record_Writer::Record_Writer(std::tr1::function<void (const byte[], size_t)> out) :
+Record_Writer::Record_Writer(std::function<void (const byte[], size_t)> out) :
    output_fn(out),
    buffer(DEFAULT_BUFFERSIZE)
    {
diff -Naur a/src/ssl/tls_client.cpp b/src/ssl/tls_client.cpp
--- a/src/ssl/tls_client.cpp	2018-02-05 15:41:49.018313394 +0100
+++ b/src/ssl/tls_client.cpp	2018-02-05 15:53:14.660325737 +0100
@@ -81,8 +81,8 @@
 /**
 * TLS Client Constructor
 */
-TLS_Client::TLS_Client(std::tr1::function<size_t (byte[], size_t)> input_fn,
-                       std::tr1::function<void (const byte[], size_t)> output_fn,
+TLS_Client::TLS_Client(std::function<size_t (byte[], size_t)> input_fn,
+                       std::function<void (const byte[], size_t)> output_fn,
                        const TLS_Policy& policy,
                        RandomNumberGenerator& rng) :
    input_fn(input_fn),
diff -Naur a/src/ssl/tls_client.h b/ssl/tls_client.h
--- a/src/ssl/tls_client.h	2018-02-05 15:41:49.018313394 +0100
+++ b/src/ssl/tls_client.h	2018-02-05 15:53:14.661325737 +0100
@@ -33,8 +33,8 @@
       void add_client_cert(const X509_Certificate& cert,
                            Private_Key* cert_key);
 
-      TLS_Client(std::tr1::function<size_t (byte[], size_t)> input_fn,
-                 std::tr1::function<void (const byte[], size_t)> output_fn,
+      TLS_Client(std::function<size_t (byte[], size_t)> input_fn,
+                 std::function<void (const byte[], size_t)> output_fn,
                  const TLS_Policy& policy,
                  RandomNumberGenerator& rng);
 
@@ -51,7 +51,7 @@
       void read_handshake(byte, const MemoryRegion<byte>&);
       void process_handshake_msg(Handshake_Type, const MemoryRegion<byte>&);
 
-      std::tr1::function<size_t (byte[], size_t)> input_fn;
+      std::function<size_t (byte[], size_t)> input_fn;
 
       const TLS_Policy& policy;
       RandomNumberGenerator& rng;
diff -Naur a/src/ssl/tls_record.h b/src/ssl/tls_record.h
--- a/src/ssl/tls_record.h	2018-02-05 15:41:49.018313394 +0100
+++ b/src/ssl/tls_record.h	2018-02-05 15:53:14.661325737 +0100
@@ -17,7 +17,7 @@
 
 #if defined(BOTAN_USE_STD_TR1)
 
-#if defined(BOTAN_BUILD_COMPILER_IS_MSVC)
+#if defined(BOTAN_BUILD_COMPILER_IS_MSVC) || defined(BOTAN_BUILD_COMPILER_IS_CLANG)
     #include <functional>
 #else
     #include <tr1/functional>
@@ -31,7 +31,7 @@
 
 namespace Botan {
 
-using namespace std::tr1::placeholders;
+using namespace std::placeholders;
 
 /**
 * TLS Record Writer
@@ -52,7 +52,7 @@
 
       void reset();
 
-      Record_Writer(std::tr1::function<void (const byte[], size_t)> output_fn);
+      Record_Writer(std::function<void (const byte[], size_t)> output_fn);
 
       ~Record_Writer() { delete mac; }
    private:
@@ -60,7 +60,7 @@
       void send_record(byte type, byte major, byte minor,
                        const byte input[], size_t length);
 
-      std::tr1::function<void (const byte[], size_t)> output_fn;
+      std::function<void (const byte[], size_t)> output_fn;
       Pipe cipher;
       MessageAuthenticationCode* mac;
 
diff -Naur a/src/ssl/tls_server.cpp b/src/ssl/tls_server.cpp
--- a/src/ssl/tls_server.cpp	2018-02-05 15:41:49.018313394 +0100
+++ b/src/ssl/tls_server.cpp	2018-02-05 15:56:39.336329421 +0100
@@ -85,8 +85,8 @@
 /*
 * TLS Server Constructor
 */
-TLS_Server::TLS_Server(std::tr1::function<size_t (byte[], size_t)> input_fn,
-                       std::tr1::function<void (const byte[], size_t)> output_fn,
+TLS_Server::TLS_Server(std::function<size_t (byte[], size_t)> input_fn,
+                       std::function<void (const byte[], size_t)> output_fn,
                        const TLS_Policy& policy,
                        RandomNumberGenerator& rng,
                        const X509_Certificate& cert,
diff -Naur a/src/ssl/tls_server.h b/src/ssl/tls_server.h
--- a/src/ssl/tls_server.h	2018-02-05 15:41:49.018313394 +0100
+++ b/src/ssl/tls_server.h	2018-02-05 15:56:14.019328966 +0100
@@ -36,8 +36,8 @@
       * FIXME: support cert chains (!)
       * FIXME: support anonymous servers
       */
-      TLS_Server(std::tr1::function<size_t (byte[], size_t)> input_fn,
-                 std::tr1::function<void (const byte[], size_t)> output_fn,
+      TLS_Server(std::function<size_t (byte[], size_t)> input_fn,
+                 std::function<void (const byte[], size_t)> output_fn,
                  const TLS_Policy& policy,
                  RandomNumberGenerator& rng,
                  const X509_Certificate& cert,
@@ -53,7 +53,7 @@
 
       void process_handshake_msg(Handshake_Type, const MemoryRegion<byte>&);
 
-      std::tr1::function<size_t (byte[], size_t)> input_fn;
+      std::function<size_t (byte[], size_t)> input_fn;
 
       const TLS_Policy& policy;
       RandomNumberGenerator& rng;
