--- a/mimetic/os/mmfile.cxx
+++ b/mimetic/os/mmfile.cxx
@@ -57,7 +57,7 @@
 bool MMFile::map()
 {
     m_beg = (char*) mmap(0, m_st.st_size, PROT_READ, MAP_SHARED,m_fd,0);
-    if(m_beg > 0)
+    if(m_beg > (char *)0)
     {
         m_end = m_beg + m_st.st_size;
         #if HAVE_MADVISE

--- a/test/cutee.cxx
+++ b/test/cutee.cxx
@@ -343,11 +343,11 @@
 
         _( "" );
         _( RUNTEST_NAME ".cxx: cutee" );
-        _( "\t$(CUTEE) -m -o "RUNTEST_NAME".cxx" );
+        _( "\t$(CUTEE) -m -o " RUNTEST_NAME ".cxx" );
         _( "" );
-        _( RUNTEST_NAME": autocutee.mk " RUNTEST_NAME ".o $(object_files)");
-        _( "\t$(CXX) $(CXXFLAGS) $(LDFLAGS) -o "RUNTEST_NAME" "RUNTEST_NAME".o $(object_files)");
-        _( "\t./"RUNTEST_NAME );
+        _( RUNTEST_NAME ": autocutee.mk " RUNTEST_NAME ".o $(object_files)");
+        _( "\t$(CXX) $(CXXFLAGS) $(LDFLAGS) -o " RUNTEST_NAME " " RUNTEST_NAME ".o $(object_files)");
+        _( "\t./" RUNTEST_NAME );
         _( "" );
         _( "# cutee autogen: end ");
     }
@@ -410,13 +410,13 @@
 
         _( "" );
         _( RUNTEST_NAME "-clean:");
-        _( "\trm -f autocutee.mk cutee *.o *.cutee.cxx "RUNTEST_NAME" "RUNTEST_NAME".cxx");
+        _( "\trm -f autocutee.mk cutee *.o *.cutee.cxx " RUNTEST_NAME " " RUNTEST_NAME ".cxx");
         _( "\ttouch autocutee.mk");
 
         _( "" );
-        _( "EXTRA_PROGRAMS="RUNTEST_NAME );
-        _( RUNTEST_NAME "_SOURCES="RUNTEST_NAME".cxx $(test_files) $(t_runners)");
-        _( RUNTEST_NAME"_DEPENDENCIES=cutee autocutee.mk" );
+        _( "EXTRA_PROGRAMS=" RUNTEST_NAME );
+        _( RUNTEST_NAME "_SOURCES=" RUNTEST_NAME ".cxx $(test_files) $(t_runners)");
+        _( RUNTEST_NAME "_DEPENDENCIES=cutee autocutee.mk" );
         _( "# cutee autogen: end ");
     }
 private:
