diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2018-05-03 11:24:46.437156894 +0200
+++ b/CMakeLists.txt	2018-05-03 11:24:05.422156155 +0200
@@ -86,10 +86,10 @@
     ENDIF()
 
     message(STATUS "Compiler info: ${CMAKE_CXX_COMPILER_ID} (${CMAKE_CXX_COMPILER}) ; version: ${CMAKE_CXX_COMPILER_VERSION}")
-    IF ( CYGWIN OR (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 5.0))
-        ADD_DEFINITIONS( -std=gnu++98 ) # to support snprintf
+    IF (( CYGWIN OR (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 5.0)) AND NOT (CMAKE_CXX_COMPILER_ID STREQUAL "Clang"))
+        SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++98" ) # to support snprintf
     ELSE()
-        ADD_DEFINITIONS( -std=c++98 )
+        SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++98" )
     ENDIF()
 
 ENDIF( MINGW OR UNIX )
