# see https://bugs.llvm.org/show_bug.cgi?id=16404

--- a/configure.ac	2016-12-31 22:53:11.000000000 +0100
+++ b/configure.ac	2016-12-31 22:53:11.000000000 +0100
@@ -221,6 +221,8 @@
 AS_CASE([$host], [*-*-os2*], [OS2_LDFLAGS="-Zargs-resp"])
 AC_SUBST([OS2_LDFLAGS])
 
+AC_SUBST(CLANG_LDFLAGS, "--rtlib=compiler-rt")
+
 AC_CONFIG_FILES([Makefile
                  doc/Makefile
                  lib/Makefile

--- a/configure	2016-12-31 23:36:36.000000000 +0100
+++ b/configure	2016-12-31 23:36:36.000000000 +0100
@@ -637,6 +637,7 @@
 LTLIBOBJS
 LIBOBJS
 OS2_LDFLAGS
+CLANG_LDFLAGS
 LIBTESTS_LIBDEPS
 LIBM4_LTLIBDEPS
 LIBM4_LIBDEPS
@@ -41314,6 +41315,7 @@
      ;;
 esac
 
+CLANG_LDFLAGS="--rtlib=compiler-rt"
 
 ac_config_files="$ac_config_files Makefile doc/Makefile lib/Makefile src/Makefile tests/Makefile checks/Makefile examples/Makefile"
 
--- a/Makefile.in	2016-12-31 23:25:57.000000000 +0100
+++ b/Makefile.in	2016-12-31 23:25:57.000000000 +0100
@@ -1186,6 +1186,7 @@
 NEXT_WCTYPE_H = @NEXT_WCTYPE_H@
 OBJEXT = @OBJEXT@
 OS2_LDFLAGS = @OS2_LDFLAGS@
+CLANG_LDFLAGS = @CLANG_LDFLAGS@
 PACKAGE = @PACKAGE@
 PACKAGE_BUGREPORT = @PACKAGE_BUGREPORT@
 PACKAGE_NAME = @PACKAGE_NAME@

--- a/src/Makefile.am   2016-12-29 21:32:18.000000000 +0100
+++ b/src/Makefile.am   2016-12-29 21:32:18.000000000 +0100
@@ -22,7 +22,7 @@
 AUTOMAKE_OPTIONS = nostdinc
 AM_CPPFLAGS = -I$(top_srcdir)/lib -I../lib
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
-AM_LDFLAGS = $(OS2_LDFLAGS)
+AM_LDFLAGS = $(OS2_LDFLAGS) ${CLANG_LDFLAGS}
 bin_PROGRAMS = m4
 noinst_HEADERS = m4.h
 m4_SOURCES = m4.c builtin.c debug.c eval.c format.c freeze.c input.c \

--- a/src/Makefile.in	2016-12-31 23:25:58.000000000 +0100
+++ b/src/Makefile.in	2016-12-31 23:25:58.000000000 +0100
@@ -1145,6 +1145,7 @@
 NEXT_WCTYPE_H = @NEXT_WCTYPE_H@
 OBJEXT = @OBJEXT@
 OS2_LDFLAGS = @OS2_LDFLAGS@
+CLANG_LDFLAGS = @CLANG_LDFLAGS@
 PACKAGE = @PACKAGE@
 PACKAGE_BUGREPORT = @PACKAGE_BUGREPORT@
 PACKAGE_NAME = @PACKAGE_NAME@
@@ -1468,7 +1469,7 @@
 AUTOMAKE_OPTIONS = nostdinc
 AM_CPPFLAGS = -I$(top_srcdir)/lib -I../lib
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
-AM_LDFLAGS = $(OS2_LDFLAGS)
+AM_LDFLAGS = $(OS2_LDFLAGS) $(CLANG_LDFLAGS)
 noinst_HEADERS = m4.h
 m4_SOURCES = m4.c builtin.c debug.c eval.c format.c freeze.c input.c \
 macro.c output.c path.c symtab.c
